app/services/store_sp_metadata_in_session.rb

```diff
@@ -0,0 +1,27 @@
+class StoreSpMetadataInSession
```

Comment:
I don't mind verb names in service class methods. I do feel like some of the service classes in this app could perhaps be classified as something else...eg `RandomPhrase`...but that's for a separate PR :)